### PR TITLE
Docs 8452 - Regression Tests

### PR DESCRIPTION
**Pull request by madelinezec**
Would love your CR when you have a moment. Hoping to get this merged tomorrow so I can test my comparison functions in stitch

### CODE REVIEW

File: worker/jobTypes/githubPushJob.js [hunk 1]
```diff
@@ -119,6 +119,16 @@ async function runGithubPush(currentJob) {
     throw invalidJobDef;
   }
 
+  // master branch cannot run through staging build
+  if (currentJob.payload.branchName === 'master' && currentJob.title != 'Regression Test') {
```
Comment (danielborowski) [inline worker/jobTypes/githubPushJob.js]: doesn't matter here but for future try and use !==

File: worker/jobTypes/regressionTestJob.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  try {
+    reposApprovedForTesting = JSON.parse(fileContents)["repos"];
+  } catch (error) {
+    const errorParsingJson = new Error("error parsing json: ", error);
+  }
+
+  /*open change stream so we can make sure we are monitoring all regression jobs */
+  return new Promise(async function(resolve, reject) {
+    const result = await regressionUtils.monitorAndCreateChildJobs(currentJob, reposApprovedForTesting);
+    if (result === true) {
```
Comment (danielborowski) [inline worker/jobTypes/regressionTestJob.js]: can just be `if (result) {`

File: worker/utils/regressionTestUtils.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+// Get username password credentials
+const username = encodeURIComponent(EnvironmentClass.getAtlasUsername());
+const password = encodeURIComponent(EnvironmentClass.getAtlasPassword());
+
+async function insertJob(payloadObj) {
+  // create the new job document
+  const newJob = {
+    title: "Regression Test",
+    user: "madelinezec",
+    email: "mez2113@columbia.edu",
```
Comment (danielborowski) [inline worker/utils/regressionTestUtils.js]: hardcoded

File: worker/utils/regressionTestUtils.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+function evaluateJobArrays(reposApprovedForTesting, completedJobs) {
+  let completedJobCounter = 0;
+  let arraysAreEqual = false;
+
+  reposApprovedForTesting.forEach(function(repository) {
+    console.log(repository.name);
+    if (completedJobs.includes(repository.name)) {
+      completedJobCounter++;
+      if (completedJobCounter === reposApprovedForTesting.length) {
+        arraysAreEqual = true;
```
Comment (danielborowski) [inline worker/utils/regressionTestUtils.js]: you can also `break` here once this is set to true to stop the loop

File: worker/utils/regressionTestUtils.js [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+      );
+
+      let completedChildJobs = [];
+      changeStream.on("change", (updatedJob, error) => {
+        if (error) {
+          console.log("error ", error);
+          throw error;
+        }
+        //def set a var here
+        var strign1  = updatedJob.fullDocument.payload.parentID.toString();
```
Comment (danielborowski) [inline worker/utils/regressionTestUtils.js]: typo

File: worker/utils/regressionTestUtils.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+        var str2 = currentJob._id.toString();
```
Comment (danielborowski) [inline worker/utils/regressionTestUtils.js]: are these used anywhere?

File: worker/jobTypes/productionDeployJob.js [hunk 7]
```diff
@@ -8,10 +8,10 @@ const buildTimeout = 60 * 450;
 const invalidJobDef = new Error('job not valid');
 
 async function verifyUserEntitlements(currentJob) {
-  const user = currentJob.user;
+  const [user] = currentJob.user;
```
Comment (skerschb) [inline worker/jobTypes/productionDeployJob.js:11]: I'm confused about the use of destructuring here -- have we changed the typing around these objects?

**Pull request merged**